src/query.c

```diff
@@ -1796,11 +1796,8 @@ static int QueryNode_CheckIsValid(QueryNode *n, IndexSpec *spec, RSSearchOptions
       break;
     case QN_VECTOR:
       if (validationFlags & QAST_DISABLE_VECTOR_QUERIES) {
-        if (n->vn.vq->type == VECSIM_QT_KNN) {
-          QueryError_SetError(status, QUERY_ESYNTAX, "VECTOR KNN queries are not allowed in a vector filter");
-          res = REDISMODULE_ERR;
-        } else if (n->vn.vq->type == VECSIM_QT_RANGE) {
-          QueryError_SetError(status, QUERY_ESYNTAX, "VECTOR RANGE queries are not allowed in a vector filter");
+        if ((n->vn.vq->type == VECSIM_QT_KNN) || (n->vn.vq->type == VECSIM_QT_RANGE)) {
```

Comment:
Since these are the two only cases, this condition will always be evaluated to `true` no?
I think you can throw the error once we find a `QN_VECTOR` node

Reply:
You are right. Done.